### ISSUE TITLE
Strange behavior of Emacs-nox in Windows Terminal decompressing files

### ISSUE THREAD

**Issue by angelog0** [msys2/MSYS2-packages#4805]
### Description / Steps to reproduce the issue

1. Create a tarball (say, `logs.tar.gz`) containing, for example, two text files 
2. Start Windows Terminal (Preview here) with just one tab and configured as you suggest for MSYS2 ("MSYSTEM": "MSYS")
3. Run `emacs logs.tar.gz`. The file is decompressed and you can work with those two text files. Quit Emacs
4. Now open a second tab in WT and re-run step 3 in the **first** tab: it prints `Error while executing "gzip -c -q -d < /home/angelo/logs.tar.gz"` in a `*jka-compr-error*` buffer. Quit Emacs
5. Repeat step 3 in the **second** tab: all works as expected in step 3.
6.  If you open more tabs, Emacs works as expected only if started from the last opened tab.

Emacs in the above steps is a build from master exactly as you build Emacs 29.4 for MSYS2 (no MINGW/UCRT)

### Expected behavior

Emacs should work from any tab. Or not?

### Actual behavior

See the previous steps

In short Emacs decompresses files only if it started from the last opened tab. 

All the above is valid also for other compression format, for example `foo.bz2` etc.

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MSYS_NT-10.0-19045

**Comment by Biswa96** [comment]
Can the issue be reproduced in mintty?

**Comment by angelog0** [comment]
But MinTTY does not have tabs! Even using its virtual tabs, really they are instances of MinTTY.

If I start two instances of WT, Emacs works as expected...

**Comment by angelog0** [comment]
Just for the record... The same issue occurs if one starts Emacs with `q/Q` options

```
emacs -q logs.tar.gz

## or

emacs -Q logs.tar.gz
```

**Comment by angelog0** [comment]
After the last build from master the issue cannot be reproduced. So I close this issue.
